#! /bin/sh /usr/share/dpatch/dpatch-run
## disable-online-updates.dpatch by Diego Iastrubni <diego.iastrubni@xorcom.com>
##
## DP: Disable online updates, unless user specifies a special hidden
## DP: configuration in amportal.conf

@DPATCH@
diff -urNad upstream~/amp_conf/htdocs/admin/page.modules.php upstream/amp_conf/htdocs/admin/page.modules.php
--- upstream~/amp_conf/htdocs/admin/page.modules.php	2007-01-04 20:31:02.000000000 +0200
+++ upstream/amp_conf/htdocs/admin/page.modules.php	2007-01-14 12:46:50.000000000 +0200
@@ -6,11 +6,16 @@
  * Optionally, the other way is to remove the below lines, and instead just define 
  * EXTERNAL_PACKAGE_MANAGEMENT as 1. This prevents changing the setting from amportal.conf.
  */
+
+/*
+on debian we force the usage of the package managment. 
 if (!isset($amp_conf['AMPEXTERNPACKAGES']) || ($amp_conf['AMPEXTERNPACKAGES'] != 'true')) {
 	define('EXTERNAL_PACKAGE_MANAGEMENT', 0);
 } else {
 	define('EXTERNAL_PACKAGE_MANAGEMENT', 1);
 }
+*/
+define('EXTERNAL_PACKAGE_MANAGEMENT', 1);
 
 
 $extdisplay = isset($_REQUEST['extdisplay'])?$_REQUEST['extdisplay']:'';
@@ -28,6 +33,15 @@
 	uninstall - uninstall local module
 */
 
+if (isset($amp_conf["AMP_ONLINE_UPDATE"])) {
+	$packager_mode = (strtolower($amp_conf["AMP_ONLINE_UPDATE"]) == "true" );
+}
+else{
+	$packager_mode =  true;
+}
+
+
+
 function pageReload(){
 return "";
 	//return "<script language=\"Javascript\">document.location='".$_SERVER['PHP_SELF']."?".$_SERVER['QUERY_STRING']."&foo=".rand()."'</script>";
